tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -166,47 +177,190 @@ class RangeDatasetOpTest : public OpsTestBase {
                               end_of_sequence);
   }
 
+  const string& name() const { return dataset_->name(); }
+
  protected:
   DatasetBase* dataset_;
-  FunctionLibraryRuntime* flr_; // not owned
+  FunctionLibraryRuntime* flr_;  // not owned
   std::function<void(std::function<void()>)> runner_;
   std::unique_ptr<DeviceMgr> device_mgr_;
   std::unique_ptr<FunctionLibraryDefinition> lib_def_;
   std::unique_ptr<ProcessFunctionLibraryRuntime> pflr_;
   std::unique_ptr<thread::ThreadPool> thread_pool_;
   std::unique_ptr<IteratorContext> iterator_context_;
   std::unique_ptr<IteratorBase> iterator_;
+  std::unique_ptr<SerializationContext> serialization_context_;
+  const string node_name_ = "range_dataset";
+  const string op_name_ = "RangeDataset";
 };
 
-TEST_F(RangeDatasetOpTest, GetNext) {
+#define RUN_TEST_GETNEXT(SUBTESTNAME, START, END, STEP)                    \
```

Comment:
use parameterized tests for this (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/grappler/optimizers/data/noop_elimination_test.cc for an example)